src/components/BookmarkPopover.tsx

```diff
@@ -10,79 +11,135 @@ import {
   Transition,
 } from "solid-headless";
 import { RiSystemAddFill } from "solid-icons/ri";
-import type { CardCollectionDTO, ScoreCardDTO } from "../../utils/apiTypes";
+import type {
+  CardBookmarksDTO,
+  CardCollectionDTO,
+  ScoreCardDTO,
+} from "../../utils/apiTypes";
 import InputRowsForm from "./Atoms/InputRowsForm";
 
 export interface BookmarkPopoverProps {
   card: ScoreCardDTO;
   cardCollections: CardCollectionDTO[];
   fetchCardCollections: () => void;
+  setLoginModal: Setter<boolean>;
 }
 
 const BookmarkPopover = (props: BookmarkPopoverProps) => {
   const apiHost = import.meta.env.PUBLIC_API_HOST;
 
   const [showCollectionForm, setShowCollectionForm] = createSignal(false);
-
+  const [notLoggedIn, setNotLoggedIn] = createSignal(false);
   const [collectionFormTitle, setCollectionFormTitle] = createSignal("");
+  const [cardCollections, setCardCollections] =
+    createSignal<CardBookmarksDTO[]>();
+  const [usingPanel, setUsingPanel] = createSignal(false);
 
+  const fetchCollections = () => {
+    void fetch(
+      `${apiHost}/card_collection/bookmark/${props.card.metadata.id}`,
+      {
+        method: "GET",
+        credentials: "include",
+      },
+    ).then((response) => {
+      if (response.ok) {
+        void response.json().then((collection) => {
+          setCardCollections(collection as CardBookmarksDTO[]);
+        });
+      }
+      if (response.status == 401) {
+        setNotLoggedIn(true);
```

Comment:
the issue is that createEffect gets run on component mount which happens when the page is loaded on not when the collections button is clicked. So, on load I j have it set a signal that the user isn't logged in so that when the button is clicked, it'll open the modal